### DISCUSSION TITLE
Document.evaluate returns null result in correct script

### DISCUSSION THREAD

**Discussion by assopri** [cefsharp/CefSharp#3557] [General]
Hello! Why script executing successfully in Chrome console returns null result in EvaluateScriptAsync (no E) ?

![image](https://user-images.githubusercontent.com/73235464/118231191-6dcaca00-b497-11eb-96da-4b0fe4b15939.png)

At the same time - if I use smth like this - works completely OK

![image](https://user-images.githubusercontent.com/73235464/118231227-7a4f2280-b497-11eb-89f9-8b981f7f5705.png)

**Comment by amaitland** [comment]
Please no images of code, I cannot copy and paste an image.

> You cannot directly return a DOM Element (or anything with a cyclic reference), you need to create a new object that has just the information you need to return.

As per https://github.com/cefsharp/CefSharp/wiki/General-Usage#2-how-do-you-call-a-javascript-method-that-returns-a-result

You'll need to convert your nodes into a tree structure rather than the default which is a graph and cannot be returned.

In future please use discussions to ask questions rather than open an issue.

**Reply by assopri** [reply]
Thank you for your help! Sorry for my rules misunderstanding!
